python/cudf/cudf/tests/test_multiindex.py

```diff
@@ -1340,3 +1340,110 @@ def test_multiIndex_argsort(pdi, ascending):
     actual = gdi.argsort(ascending=ascending)
 
     assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize(
+    "idx", [pd.MultiIndex.from_product([["python", "cobra"], [2018, 2019]])]
+)
+@pytest.mark.parametrize(
+    "names", [[None, None], ["a", None], ["new name", "another name"]]
+)
+@pytest.mark.parametrize("inplace", [True, False])
+def test_multiindex_set_names(idx, names, inplace):
+    pi = idx.copy()
+    gi = cudf.from_pandas(idx)
+
+    expected = pi.set_names(names=names, inplace=inplace)
+    actual = gi.set_names(names=names, inplace=inplace)
+
+    if inplace:
+        expected, actual = pi, gi
+
+    assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize(
+    "idx",
+    [
+        pd.MultiIndex.from_product(
+            [["python", "cobra"], [2018, 2019], ["aab", "bcd"]]
+        ),
+        pd.MultiIndex.from_product(
+            [["python", "cobra"], [2018, 2019], ["aab", "bcd"]],
+            names=["one", "two", "three"],
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "level, names",
+    [
+        (0, "abc"),
+        (1, "xyz"),
+        ([2, 1], ["a", "b"]),
+        (None, ["a", "b", "c"]),
+        (None, ["a", None, "c"]),
+    ],
+)
+@pytest.mark.parametrize("inplace", [True, False])
+def test_multiindex_set_names_level_with_integers(idx, level, names, inplace):
+    pi = idx.copy()
+    gi = cudf.from_pandas(idx)
+
+    expected = pi.set_names(names=names, level=level, inplace=inplace)
+    actual = gi.set_names(names=names, level=level, inplace=inplace)
+
+    if inplace:
+        expected, actual = pi, gi
+
+    assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize(
+    "idx",
+    [
+        pd.MultiIndex.from_product(
+            [["python", "cobra"], [2018, 2019], ["aab", "bcd"]],
+            names=["one", None, "three"],
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "level, names",
+    [
+        ([None], "abc"),
+        (["three", "one"], ["a", "b"]),
+        (["three", 1], ["a", "b"]),
+        ([0, "three", 1], ["a", "b", "z"]),
+        (["one", 1, "three"], ["a", "b", "z"]),
+        (["one", None, "three"], ["a", "b", "z"]),
```

Comment:
I hate to ask this... but what's the behavior is someone passes in say `[0, 1]` and the multiindex levels are named `[1, 0]`? Does Pandas resolve to using integers as positions or names first? Should we test this?

Reply:
> Does Pandas resolve to using integers as positions or names first? 

Pandas tries to resolve the input first as labels and then as positions. Similar to what our `MultiIndex._level_index_from_level` does.

> Should we test this?

Yep, added tests for this in `test_multiindex_set_names_default_and_int_names`